### PR TITLE
feat(infer): add max_chunk_seconds option

### LINKED ISSUE #183: Split chunk if it's too long
When inferencing files that have long prolonged sounds, it causes CUDA Out of Memory error because the chunks are too long.

If the chunk is very large, split it into two or more smaller chunks and process them instead.
It would probably cause a gap or a click sound between two chunks then maybe do a bit of cross-fading.

### PR DESCRIPTION
**Pull request by maximxlss**
add max_chunk_seconds, option that helps split long chunks to avoid OOM

- Closes #183
- Currently simply cuts chunks into smaller pieces. Crossfade and smarter cutting might help, but it currently doesn't seem too bad.
- I have chosen a default value of 40 seconds, extrapolating my 24 gb card handling about 150 seconds to the base 6 gb. In reality it might be not the actual limit on 6 gb, needs testing.
- Could be off by default, but I think this is a fair limitation to prevent unexpected OOM errors.

### CODE REVIEW

**Comment by maximxlss** [comment]
Sorry for the noob questions, basically my first actual feature pull request lol:

- Have just pressed some button and made this eeb7ecd. Is it okay? Should I do something?
- Checks fail cuz of formatting. Should I amend the commit or make a new one? What should be the commit message if I make a new one?

**Comment by 34j** [comment]
Too perfect. I think amend is the legit way to do it, but you can do it either way. （If you don't amend, style: run pre-commit or something?)

File: src/so_vits_svc_fork/__main__.py [hunk 1]
```diff
@@ -202,6 +202,13 @@ def train(
     default=False,
     help="absolute thresh",
 )
+@click.option(
+    "-mc",
+    "--max_chunk_seconds",
```
Comment (34j) [inline src/so_vits_svc_fork/__main__.py:207]:
```suggestion
    "--max-chunk-seconds",
```

**Review by 34j — CHANGES_REQUESTED** [review]

**Comment by 34j** [comment]
@allcontributors add maximxlss code

**Pull request merged**